engine/class_modules/sc_death_knight.cpp

```diff
@@ -4524,7 +4545,7 @@ struct death_and_decay_base_t : public death_knight_spell_t
 struct death_and_decay_t : public death_and_decay_base_t
 {
   death_and_decay_t( death_knight_t* p, const std::string& options_str ) :
-    death_and_decay_base_t( p, "death_and_decay", p -> spec.death_and_decay )
+    death_and_decay_base_t( p, p -> covenant.deaths_due -> ok() ? "deaths_due" : "death_and_decay", p -> spec.death_and_decay )
```

Comment:
Is the ability death and decay (from spec.death_and_decay) or death's due (covenant.deaths_due) with the covenant active?

Reply:
It's an override.  Looks similar to how defile is handled.  It does have it's own spell, with it's own spell data, but everything matches, and even stuff within death's due is applied to death and decay.  Like the obliterate cleave as mentioned below.

I honestly wasn't sure what was going to be the cleanest implementation for these.  death's due could have been implemented as it's own, basically as a copy of death and decay, but we would still have to apply the manual overrides to defile.  This whole thing looks kinda like a mess, so I wouldn't be shocked to see it get an overhaul at some point.  Or it may stay this way all expansion since it will be tossed out in the next one.

Name             : Death's Due (desc=Night Fae) (id=315442) [Spell Family (15), Passive]
Class            : Death Knight
School           : Nature
Spell Type       : None
Attributes       : ......x. ........ ........ ........   ........ ........ ........ ........
                 : ........ ........ ........ ........   ........ ........ ........ ........
                 : ........ .......x ........ ........   ........ ........ ........ ........
                 : ........ ........ ........ ........   ........ ........ ........ ........
                 : ......x. ....x... ........ ........   ........ ........ ........ ........
                 : ........ ........ ........ ........   ........ ........ ........ ........
                 : ........ ........ ........ ........   ........ ........ ........ ........
                 : ........ ........ ........ ........
                 : Passive (6)
Effects          :
#1 (id=799201)   : Apply Aura (6) | Override Action Spell (Misc /w Base) (332)
                   Base Value: 324128 | Scaled Value: 324128 | PvP Coefficient: 1.00000 | Misc Value: 43265 | Misc Value 2: 0x3 | Target: Self (1)
#2 (id=871412)   : Apply Aura (6) | Add Flat Modifier (107): Spell Effect 4 (32)
                   Base Value: 1 | Scaled Value: 1 | PvP Coefficient: 1.00000 | Misc Value: 32 | Target: Self (1)
                   Affected Spells: Death and Decay (188290)
                   Family Flags: 118
Description      : Learn Death's Due, replacing Death and Decay.

Tooltip          : Drain power from each enemy within the area, reducing their damage done by 10% and increasing your Strength by 80 for 10 sec.

Reply:
I was just wondering if it's a good idea to keep using spec.death_and_decay for the spell's data instead of covenant.deaths_due, since it is, I assume, the actual ability used by the player.